service/javadsl/server/src/main/scala/com/lightbend/lagom/internal/javadsl/server/JavadslServerBuilder.scala

```diff
@@ -67,25 +69,31 @@ class JavadslServerBuilder @Inject() (environment: Environment, httpConfiguratio
   }
 
   /**
-   * Create a service info for the given interface.
-   *
-   * @param interface The interface to create the service info for.
-   * @return The service info.
+   * Create a service info for the given interfaces.
    */
-  def createServiceInfo(interface: Class[_]): ServiceInfo = {
-    if (classOf[Service].isAssignableFrom(interface)) {
-      val descriptor = ServiceReader.readServiceDescriptor(
-        environment.classLoader,
-        interface.asSubclass(classOf[Service])
-      )
-      new ServiceInfo(descriptor.name())
+  def createServiceInfo(primaryServiceInterface: Class[_], secondaryServices: Seq[Class[_]]): ServiceInfo = {
+    val interfaces = primaryServiceInterface +: secondaryServices
+    if (interfaces.forall(classOf[Service].isAssignableFrom)) {
+      val descriptors = interfaces.map { serviceInterface =>
+        ServiceReader.readServiceDescriptor(
+          environment.classLoader,
+          serviceInterface.asSubclass(classOf[Service])
+        )
+      }
+      val locatableServices = descriptors
+        .filter(_.locatableService())
+        .map { descriptor =>
+          descriptor.name() -> TreePVector.from(descriptor.acls().stream().collect(Collectors.toList()))
+        }.toMap.asJava
+      new ServiceInfo(descriptors.head.name, HashTreePMap.from(locatableServices))
     } else {
-      throw new IllegalArgumentException(s"Don't know how to load services that don't implement Service: $interface")
+      throw new IllegalArgumentException(s"Don't know how to load services that don't implement Service. Provided: ${interfaces.mkString("[", ", ", "]")}")
```

Comment:
I think we can make all of this more type safe:

1. Change the type parameter of `ServiceBinding` and everything that it affects to `T extends Service`
2. Change `ServiceInfoProvider` to take `Class[_ <: Service]`
3. Change this method to do the same

Then the dynamic type check isn't really necessary.

This would technically break source compatibility, but only for cases that wouldn't have worked in the first place (calling `bindServices` on things that aren't service interfaces).

I tried this out in one of my Chirper projects... calling `serviceBinding(List.class, ArrayList.class)` compiles currently and fails on startup, but not with the changes I'm proposing it fails at compile time.

Happy to do this in a separate PR if you prefer.